### ISSUE TITLE
[@typescript-eslint/typedef] In the class, even if there are types, there will still check not passed

### ISSUE THREAD

**Issue by duan602728596** [typescript-eslint/typescript-eslint#4033]
- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [read the FAQ](https://github.com/typescript-eslint/typescript-eslint/blob/master/docs/getting-started/linting/FAQ.md) and my problem is not listed.

**Repro**

```JSON
{
  "rules": {
    "@typescript-eslint/typedef": [
      "error",
      {
        "arrayDestructuring": true,
        "arrowParameter": true,
        "memberVariableDeclaration": true,
        "objectDestructuring": true,
        "parameter": true,
        "propertyDeclaration": true,
        "variableDeclaration": true
      }
    ]
  }
}
```

```TS
class ClassName {
  public str: string = 'str';

  func: () => void = (): void => {
    console.log(this.a);
  };
}
```

**Expected Result**

Eslint check passed.

**Actual Result**

Line 2:
ESLint: Expected str to have a type annotation.(@typescript-eslint/typedef).
Line 4:
ESLint: Expected func to have a type annotation.(@typescript-eslint/typedef)

**Additional Info**

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `5.1.0` |
| `@typescript-eslint/parser`        | `5.1.0` |
| `TypeScript`                       | `4.4.4` |
| `ESLint`                           | `8.0.1` |
| `node`                             | `16.11.1` |

**Comment by bradzacher** [comment]
I am  unable to reproduce this against master.
